electron/libelectron/src/backends/photon.cc

```diff
@@ -19,6 +18,9 @@ void PhotonBackend::loadlib(const std::string& filename) {
 
     // TODO: This shouldn't be an assert but need to add error checking
     assert(lib_handle != nullptr);
+    //TODO: Amount of memory to allocate?
+    dev_mem_obj = new memory_mgmt::FirstFitAllocator(10*4000);
```

Comment:
This should happen in the PhotonBackend constructor:
```cpp
PhotonBackend::PhotonBackend() :
    lib_handle{nullptr},
    emu(std::make_shared<photon::PhotonEmu>()) { }
```
becomes
```cpp
PhotonBackend::PhotonBackend() :
    lib_handle{nullptr},
    emu(std::make_shared<photon::PhotonEmu>()),
    allocator(std::make_shared<memory_mgmt::FirstFitAllocator>(10*4000)) { }
```

We can figure out the size later -- for now set it to 8GB (`8 * 2 << 30`)

Reply:
I tried to add the allocator and device_memory the same way as here but had to enforce the sequential logic to pass device memory to allocator. Hence moved it within the brackets. Let me know if there is any other way.